File: projects/engineering.purdue.edu/kak/scriptingwithobjects/swocode/chap13/GetEventDescriptors.py
#!/usr/bin/env python

### GetEventDescriptors.py

from Tkinter import *                                                #(A)

mw = Tk()                                                            #(B)
b = Button( mw )                                                     #(C)

print "All system-supplied event descriptors for\n" +  \
"the Button class:\n"
print "\n".join( b.bind_class( b.winfo_class() ) ) + "\n\n"          #(D)    

def foo(): pass

# Declare additional event descriptors for the Button class:
b.bind_class( 'Button', '<Button-2>', foo )                          #(E)
b.bind_class( 'Button', '<ButtonRelease-2>', foo )                   #(F)
print "All event descriptors defined for Button class\n" + \
"including the new ones:\n"
print "\n".join( b.bind_class( b.winfo_class() ) ) + "\n\n"          #(G)    

# Declare event descriptors for a specific button instance:
b.bind( '<Button-3>', foo)                                           #(H)
b.bind( '<Button-4>', foo )                                          #(I)
b.bind( '<Button-5>', foo )                                          #(J)
b.bind( '<ButtonRelease-3>', foo )                                   #(K)
b.bind( '<ButtonRelease-4>', foo )                                   #(L)
b.bind( '<ButtonRelease-5>', foo )                                   #(M)

print "\nAll user-defined event descriptors\n" +   \
"for this specific button:\n"
print "\n".join( b.bind() ) + "\n\n"                                 #(N)    

File: projects/engineering.purdue.edu/scriptingwithobjects/swocode/chap13/GetEventDescriptors.pl
#!/usr/bin/perl -w

### GetEventDescriptors.pl

use strict;
use Tk;

my $mw = MainWindow->new();                                          #(A)
my $b = $mw->Button();                                               #(B)

print "All system-supplied event descriptors for \
the Button class:\n\n";
print join( "\n", $b->bind( ref $b ) ), "\n\n";                      #(C)    

# Declare additional event descriptors for the Button class:
$b->bind( 'Tk::Button', '<Button-2>', sub { $mw->foo2 } );           #(D)
$b->bind( 'Tk::Button', '<ButtonRelease-2>', sub { $mw->foo2 } );    #(E)

print "All event descriptors defined for Button class \
including the new ones:\n\n";
print join( "\n", $b->bind( ref $b ) ), "\n";                        #(F)    

# Declare event descriptors for a specific button instance:
$b->bind( '<Button-3>', sub { $mw->foo3 } );                         #(G)
$b->bind( '<Button-4>', sub { $mw->foo4 } );                         #(H)
$b->bind( '<Button-5>', sub { $mw->foo5 } );                         #(I)
$b->bind( '<ButtonRelease-3>', sub { $mw->bar3 } );                  #(J)
$b->bind( '<ButtonRelease-4>', sub { $mw->bar4 } );                  #(K)
$b->bind( '<ButtonRelease-5>', sub { $mw->bar5 } );                  #(L)

print "\nAll user-defined event descriptors for \
a specific button:\n\n";
print join( "\n", $b->bind ), "\n";                                  #(M)
